packages/checkbox/src/Checkbox.tsx

```diff
@@ -206,7 +205,8 @@ export default class Checkbox extends PureComponent<
   };
 
   render() {
-    const checkboxId = this.props.id || this.checkboxId;
+    const checkboxId =
+      this.props.id || `checkbox-${Checkbox.idAllocator.generate()}`;
```

Comment:
This ID generation runs on every render right now – each time the checkbox becomes checked, the ID changes which could pose accessibility problems.

Additionally, the ID seems to increment by two every time `generate()` is invoked, which seems unintentional (it's probably strict mode).